File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_element = -1
        temp = [-1]*len(prices)
        for i in range(len(prices)-1  , -1 , -1):
            max_element = max(prices[i] , max_element )
            temp[i] = max_element
        result =  0
        for i in range(len(prices)):
            result = max(temp[i] - prices[i]  , result)
        return (result)

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        Stack<int> stack = new Stack<int>();
			int MaxProfit = 0;
			for (int i = prices.Length-1; i >= 0; i--)
			{
				if (stack.Count == 0 || stack.Peek() < prices[i])
				{
					stack.Push(prices[i]);
				}
				else
				{
					int profit = stack.Peek() - prices[i];
					MaxProfit = Math.Max(profit, MaxProfit);
				}
			}
			return (MaxProfit);


    }
}